backend/api/Areas/Keycloak/Controllers/UserController.cs

```diff
@@ -1,96 +1,261 @@
+        /// </summary>
+        /// <param name="page"></param>
+        /// <param name="quantity"></param>
+        /// <param name="search"></param>
         /// <returns></returns>
         [HttpGet]
-        public async Task<IActionResult> GetUsersAsync()
+        public async Task<IActionResult> GetUsersAsync(int page = 1, int quantity = 10, string search = null)
         {
-            _optionsKeycloak.Admin.Validate();
-            var response = await _requestClient.GetAsync($"{_optionsKeycloak.Admin.Authority}{_optionsKeycloak.Admin.Users}");
+            var kusers = await _keycloakAdmin.GetUsersAsync((page - 1) * quantity, quantity, search);
+
+            // TODO: Need better performing solution.
+            var eusers = kusers.Select(u => _pimsAdminService.User.Find(u.Id) ?? new Entity.User(u.Id, $"{u.LastName}, {u.FirstName}", u.Email)
+            {
+                FirstName = u.FirstName,
+                LastName = u.LastName,
+                IsDisabled = u.Enabled
+            });
+            // TODO: Apply search to this query.
+            // TODO: Use a mapper.
+            var result = eusers.Select(u => new Model.UserModel(u));
 
-            return await response.HandleResponseAsync<IEnumerable<Model.UserModel>>();
+            return new JsonResult(result);
         }
 
         /// <summary>
         /// Fetch the user for the specified 'id'.
+        /// If the user does not exist in keycloak or PIMS return a 400-BadRequest.
         /// </summary>
+        /// <exception type="KeyNotFoundException">The user does not exist in keycloak.</exception>
         /// <returns></returns>
         [HttpGet("{id}")]
-        public async Task<IActionResult> GetUserAsync(string id)
+        public async Task<IActionResult> GetUserAsync(Guid id)
         {
-            _optionsKeycloak.Admin.Validate();
-            var response = await _requestClient.GetAsync($"{_optionsKeycloak.Admin.Authority}{_optionsKeycloak.Admin.Users}/{id}");
+            var kuser = await _keycloakAdmin.GetUserAsync(id) ?? throw new KeyNotFoundException();
+            var euser = _pimsAdminService.User.Get(kuser.Id) ?? throw new KeyNotFoundException();
+            var result = _mapper.Map<Model.UserModel>(euser);
 
-            return await response.HandleResponseAsync<IEnumerable<Model.UserModel>>();
+            return new JsonResult(result);
         }
 
         /// <summary>
         /// Update the user for the specified 'id'.
+        /// If the user does not exist in Keycloak or PIMS return a 400-BadRequest.
         /// </summary>
+        /// <exception type="KeyNotFoundException">The user does not exist in Keycloak or PIMS.</exception>
         /// <returns></returns>
         [HttpPut("{id}")]
-        public async Task<IActionResult> UpdateUserAsync(string id, [FromBody] Model.UserModel model)
+        public async Task<IActionResult> UpdateUserAsync(Guid id, [FromBody] Model.Update.UserModel model)
```

Comment:
I'm starting to wonder if it is better to have a custom model for each operation, or just one model that contains more fields then necessary. Obviously this isn't perfect as the backend would end up ignoring some fields sent in by the frontend, but it would cut done on the number of Objects/Mappers that we need.